File: leetcode/easy/202_快乐数/202_快乐数.py
# 哈希表法
class Solution:
    def getSum(self, n: int) -> int:
        sum_val = 0
        while n:
            n, digit = divmod(n, 10)
            sum_val += digit ** 2
        return sum_val
    
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1 and n not in seen:
            seen.add(n)
            n = self.getSum(n)
        return n == 1

File: leetcode/easy/202_快乐数/202_快乐数.go
// 哈希表法
func getSum(n int) int {
    sum := 0
    for n > 0 {
        sum += (n % 10) * (n % 10)
        n /= 10
    }
    return sum
}

func isHappy(n int) bool {
    m := map[int]bool{}
    for ; n != 1 && !m[n]; n, m[n] = getSum(n), true { }
    return n == 1
}